aptos-move/vm-genesis/src/lib.rs

```diff
@@ -617,10 +648,46 @@ pub fn generate_mainnet_genesis(
             voting_duration_secs: 7 * 24 * 3600, // 7 days
             voting_power_increase_limit: 30,
         },
+        initial_balances.as_slice(),
     );
     (genesis, test_validators)
 }
 
+/// Uses the staking amounts to determine what the correct initial balances
+/// if there are no other accounts that are initialized with coins
+pub fn parse_initial_balances_from_staking_amounts(
```

Comment:
I think we should just have balances separately. The staking amount can be different from a validator's total balance (e.g. due to gas). staking amount is used specifically here https://github.com/aptos-labs/aptos-core/pull/3939/files#diff-cb728fc6a391f08ea20845355e8406cc6741c7163a388a7ad5ef0e250d1d6207R177 while initial balance is now done in the new mint step you added.

Reply:
We can assertion in genesis tooling somewhere that staking amount <= initial balance

Reply:
We do have balances separately.

The purpose of this is so for the current testnets, testing frameworks, etc.  We then don't have to build up this file every place we do genesis, instead we build up the information from the already existing information.

Reply:
I'll replace this and weave through the separate ones later

Reply:
OK sounds good 👍